parsec/backend/postgresql/migrations/0010_add_pki_certificate_table.sql

```diff
@@ -0,0 +1,51 @@
+ -- Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
+
+
+-------------------------------------------------------
+--  Migration
+-------------------------------------------------------
+
+CREATE TYPE enrollment_state AS ENUM (
+    'SUBMITTED',
+    'ACCEPTED',
+    'REJECTED',
+    'CANCELLED'
+);
+
+CREATE TYPE pki_enrollment_info_accepted AS(
+    accepted_on TIMESTAMPTZ,
+    accepter_der_x509_certificate BYTEA,
+    accept_payload_signature BYTEA,
+    accept_payload BYTEA
+);
+
+CREATE TYPE pki_enrollment_info_rejected AS(
+    rejected_on TIMESTAMPTZ
+);
+
+CREATE TYPE pki_enrollment_info_cancelled AS(
+    cancelled_on TIMESTAMPTZ
+);
+
+CREATE TABLE pki_enrollment (
+    _id SERIAL PRIMARY KEY,
+    organization INTEGER REFERENCES organization (_id) NOT NULL,
+
+    enrollment_id UUID NOT NULL,
+    submitter_der_x509_certificate BYTEA NOT NULL,
+    submitter_der_x509_certificate_sha1 BYTEA NOT NULL,
+
+    submit_payload_signature BYTEA NOT NULL,
+    submit_payload BYTEA NOT NULL,
+    submitted_on TIMESTAMPTZ NOT NULL,
+
+    accepter INTEGER REFERENCES device (_id),
```

Comment:
why accepter and accepted are not part of `pki_enrollment_info_accepted` type ?

Reply:
on top of that, I would rename `accepted` into `submitter_accepted_device` (or maybe `new_device` if it is defined inside `pki_enrollment_info_accepted`)

Reply:
It's better to move it into  `pki_enrollment_info_accepted`, i tried to mimic `memory` implementation where those field are outside of the Accepted structure

Reply:
It appears that `REFERENCES` can not be used inside a `TYPE`